server.R

```diff
@@ -103,4 +103,68 @@ function(input, output, session) {
       )
     ggplotly(p)
   })
-}
\ No newline at end of file
+    
+  #********************************************************************************************************
+  # GSD start
+  #********************************************************************************************************
+  value <- reactiveValues(etaGSD = 0) # create reactive variables for dilution effect eta
+  value <- reactiveValues(psiGSD = 1) # create reactive variables for variance inflation psi
+  
+  output$dilutionSettings <- renderUI({ # show input for eta if box is unticked
+    if(input$dilutionCheck == FALSE) {
+      sliderInput("etaGSD", label=HTML("Dilution effect &eta;"), min = 0, max = 1, value = 0)
+    }
+  })
+  observeEvent(input$etaGSD, { # update eta according to slider
+    if(input$dilutionCheck == FALSE) value$etaGSD <- input$etaGSD
+  })
+  observeEvent(input$dilutionCheck, { # update eta if box is ticked again
+    if(input$dilutionCheck == TRUE) value$etaGSD <- 0
+  })
+  
+  output$psiSettings <- renderUI({ # show input for psi if box is unticked
+    if(input$psiCheck == FALSE) {
+      numericInput("psiGSD", label=HTML("Variance inflation &psi;"), min = 0.1, max = 5, value = 1, step = 0.1)

```

Comment:
Since the variance could be larger or smaller, is it better to use "Variance inflation/deflation &psi;"?